modules/common/route/route.go

```diff
@@ -88,8 +92,43 @@ func (r *Route) CreateOrPatch(
 		},
 	}
 
+	// handle possible overrides of Labels, Annotations and Spec
+	if r.override != nil {
+		if r.override.EmbeddedLabelsAnnotations != nil {
+			if r.override.Labels != nil {
+				r.route.Labels = util.MergeStringMaps(r.override.Labels, r.route.Labels)
+			}
+			if r.override.Annotations != nil {
+				r.route.Annotations = util.MergeStringMaps(r.override.Annotations, r.route.Annotations)
+			}
+		}
+		if r.override.Spec != nil {
+			originalSpec, err := json.Marshal(r.route.Spec)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("error marshalling Route Spec: %w", err)
+			}
+
+			patch, err := json.Marshal(r.override.Spec)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("error marshalling Route Spec override: %w", err)
+			}
+
+			patchedJSON, err := strategicpatch.StrategicMergePatch(originalSpec, patch, routev1.RouteSpec{})
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("error patching Route Spec: %w", err)
+			}
+
+			patchedSpec := routev1.RouteSpec{}
+			err = json.Unmarshal(patchedJSON, &patchedSpec)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("error unmarshalling patched Route Spec: %w", err)
+			}
+			r.route.Spec = patchedSpec
```

Comment:
OK so this defines what the override does. Still we need to define for the end user what that means. See my comment in the keystone patch with example questions.